scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -96,163 +117,258 @@
-                                 [:expected/completed :missing] nil ; Cause it to be deleted.
-                                 [:expected/killed :pod/waiting] (kill-task api-client expected-state-dict pod)
-                                 [:expected/killed :pod/running] (kill-task api-client expected-state-dict pod)
-
-                                 [:expected/killed :pod/succeeded] (do
-                                                                     (pod-has-just-completed kcc existing-state-dict)
-                                                                     (remove-finalization-if-set-and-delete api-client expected-state-dict pod))
-                                 [:expected/killed :pod/failed]
-                                 (do
-                                   (pod-has-just-completed kcc existing-state-dict)
-                                   (remove-finalization-if-set-and-delete api-client expected-state-dict pod))
-                                 [:expected/killed :missing] (pod-was-killed kcc pod-name)
-                                 ; TODO: Implement :pod/unknown cases.
-                                 ; TODO: Implement :pod/pending cases.
-                                 ; TODO: Implement :pod/need-to-fail cases.
-                                 [:missing :pod/running] (kill-task api-client expected-state-dict pod)
-                                 [:missing :pod/pending] (kill-task api-client expected-state-dict pod)
-                                 [:missing :pod/need-to-fail] (kill-task api-client expected-state-dict pod)
-                                 [:missing :pod/succeeded] (remove-finalization-if-set-and-delete api-client expected-state-dict pod)
-                                 [:missing :pod/failed] (remove-finalization-if-set-and-delete api-client expected-state-dict pod)
-                                 [:missing :missing] nil ; this can come up due to the recur at the end
-                                 (do
-                                   (log/error "Unexpected state: "
-                                              (vector (or expected-state :missing) (or (:state synthesized-state) :missing))
-                                              "for pod" pod-name)
-                                   expected-state-dict))]
-      (when-not (expected-state-equivalent? expected-state-dict new-expected-state-dict)
-        (update-or-delete! expected-state-map pod-name new-expected-state-dict)
+      [pod-synthesized-state-modified (or (:state synthesized-state) :missing)
+       new-cook-expected-state-dict (case (or cook-expected-state :missing)
+                                      :cook-expected-state/completed
+                                      (case pod-synthesized-state-modified
+                                        :missing nil ; Cause this entry to be deleted by update-or-delete! called later down.
```

Comment:
These comments can be before the lines?. E,g,
```
; Cause this entry to be deleted by update-or-delete! called later down.
:missing nil 
```

The same applies to all below.
This helps avoid making the lines longer than they already are.

Reply:
Ok.